### PR TITLE
7761 - custom cache context added

### PR DESCRIPTION
**Pull request by ivandudardb**
created and used a custom cache context

### CODE REVIEW

**Review by faust174 — COMMENTED** [review]
LGTM

File: web/modules/custom/custom_weather/src/Cache/CustomWeatherContext.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+namespace Drupal\custom_weather\Cache;
+
+use Drupal\Core\Cache\CacheableMetadata;
+use Drupal\Core\Cache\Context\CacheContextInterface;
+use Drupal\Core\Database\Connection;
+use Drupal\Core\Session\AccountProxyInterface;
+use Drupal\custom_weather\Service\UserCityHandler;
+
+/**
+ *
```
Comment (lavrenchuk-dev) [inline web/modules/custom/custom_weather/src/Cache/CustomWeatherContext.php:12]: *

File: web/modules/custom/custom_weather/src/Cache/CustomWeatherContext.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  public static function getLabel() {
+    return 'City cache context';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getContext() {
+    $city = $this->userCityHandler->getCurrentCity();
+    return 'custom_cache_context:' . $city;
```
Comment (lavrenchuk-dev) [inline web/modules/custom/custom_weather/src/Cache/CustomWeatherContext.php:45]: custom_comment_review

File: web/modules/custom/custom_weather/src/Cache/CustomWeatherContext.php [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+   * @param \Drupal\Core\Session\AccountProxyInterface $accountProxy
+   *   The account proxy service.
+   * @param \Drupal\Core\Database\Connection $connection
+   *   The database connection service.
+   * @param \Drupal\custom_weather\Service\UserCityHandler $userCityHandler
+   *   The user city handler service.
+   */
+  public function __construct(
+    protected AccountProxyInterface $accountProxy,
+    protected Connection $connection,
```
Comment (lavrenchuk-dev) [inline web/modules/custom/custom_weather/src/Cache/CustomWeatherContext.php:28]: is it used?

File: web/modules/custom/custom_weather/src/Cache/CustomWeatherContext.php [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+   *
+   * @param \Drupal\Core\Session\AccountProxyInterface $accountProxy
+   *   The account proxy service.
+   * @param \Drupal\Core\Database\Connection $connection
+   *   The database connection service.
+   * @param \Drupal\custom_weather\Service\UserCityHandler $userCityHandler
+   *   The user city handler service.
+   */
+  public function __construct(
+    protected AccountProxyInterface $accountProxy,
```
Comment (lavrenchuk-dev) [inline web/modules/custom/custom_weather/src/Cache/CustomWeatherContext.php:27]: is it used?

**Review by lavrenchuk-dev — CHANGES_REQUESTED** [review]

**Review by lavrenchuk-dev — APPROVED** [review]

**Pull request merged**